src/redux/saga/sagaImagesGallery.js

```diff
@@ -0,0 +1,36 @@
+import { put, call, select, takeEvery } from 'redux-saga/effects'
+
+import { imageAPI } from 'utils/axiosAPI/api'
+
+import { fetchImagesGallerySuccess, toggleIsLoading } from '../actions'
+import { REQUEST_IMAGES_GALLERY } from '../types'
+import {
+  getCurrentPage,
+  getEndYear,
+  getSearchText,
+  getStartYear,
+} from '../selectors'
+
+function* fetchImagesGallery() {
+  yield put(toggleIsLoading(true))
+
+  const searchText = yield select(getSearchText)
+  const page = yield select(getCurrentPage)
+  const yearStart = yield select(getStartYear)
+  const yearEnd = yield select(getEndYear)
```

Comment:
if you have 4 **select** operators to just call an API - that means that something is not right.
To be honest, I would combine all these 4 property into one object in reducer, then saga will look simpler like that:
```
const searchDetails = yield select(getSearchDetails)
const { searchText, page, yearStart, yearEnd } = searchDetails
```

The advantage is that you have **1** select operator, not **4**
Will that be possible to do ?